doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,293 @@
+
+#### Test Output
+
+Custom tests are expected to produce log output that conforms to the
+scorecard v1alpha2 Test Result output.  This allows scorecard to parse
+the custom test log output and aggregate the results like any other
+scorecard test.
+
+The scorecard expected output would be in JSON, and would match the following definition:  https://github.com/operator-framework/operator-sdk/blob/master/pkg/apis/scorecard/v1alpha2/types.go#L36
+
+The scorecard v1alpha2 Test Result output contains information such as
+pass or fail status and suggestions/errors produced by the test that
+the end user would want to see in the aggregated scorecard results.
+
+
+### Custom Test Development Model
+
+The developer of a custom test would have a workflow similar to:
+ 
+ * write a custom test binary in the language of their choice, sample custom test within the SDK repository would be available as a starting point
+ * test the custom test binary locally (out-of-cluster) to make sure the output is produced that matches v1alpha2 ScorecardTestResult format
+ * create a Pod manifest to run their custom test image, copying a sample Pod manifest from the scorecard examples in the SDK repository
+ * when the test binary works as expected, the developer would build and push the test image to a container registry accessible to the scorecard test environment
+ * the developer would rebuild their bundle image passing in their scorecard configuration file and their test pod manifests
+ * the Basic and OLM test manifests would be added automatically to the bundle image as part of the bundle image build process.  This allows users to run the Basic and OLM tests along with their custom tests.
+ * lastly, the developer would run the scorecard passing in their operator bundle image they just built along with any other scorecard flags (e.g. selector).
+
+### Custom Test Packaging and Configuration
+
+In this design proposal, custom tests are Pod manifests in YAML that run a custom test image.  End users would add their test pod manifests and scorecard configuration file into their operator bundle image using a CLI command developed as part of this proposal.
+
+### Scorecard Configuration
+
+Note that it would also be possible to add a resource limit to the scorecard configuration as shown in the example below.  Those resource limits would be applied to custom tests that are executed allowing for some control as to how much resources are available for the test execution.
+
+Custom test Pod manifests would be accessible from a local disk location to scorecard.  Scorecard would obtain the custom test manifests as part of its bundle image processing since the bundle image is passed into scorecard as a configuration flag.  
+
+Scorecard would assume that the scorecard configuration file and test
+manifests would be located in the bundle image at the following location:
+```
+/bundle/scorecard/.osdk-scorecard.yaml
+/bundle/scorecard/tests
+```
```

Comment:
We need to discuss the specific bundle locations with @ecordell

Reply:
so, thinking out loud here...

you might have a command like this:

operator-sdk  bundle  create --directory ./deploy/olm-catalog/memcached-operator --channels stable,beta --default-channel stable quay.io/example/test-operator:v0.1.0 **--scorecard ./scorecard** 

so, adding a flag to the above command would create a bundle image that also pulls in the scorecard config/manifests from local disk location (e.g. ./scorecard)...

to be consistent with the other bundle image LABELS you could have a LABEL like maybe:
LABEL operators.operatorframework.io.bundle.scorecard.v1=scorecard/

Reply:
https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/build.go#L53

or maybe its easier given the above impl to just store the scorecard stuff under the existing manifest directory as passed in with "--directory", if that will work, then no changes to operator-registry would be required to have 'other' directories.  With this approach, I'm assuming that adding scorecard manifests into this directory won't cause downstream problems.

Reply:
We should at least default to a well-known location within the image, perhaps enforce a specific location. I don't think configuring location within the bundle image adds much positive UX.

Reply:
I think the location _within_ the image should be defined by the bundle spec and enforced. That's what I was referring to about getting @ecordell's input.

The location in the project directory could be `<project-root>/test/scorecard` (or whatever would be a sibling of the e2e tests).